elogia_sale/models/elogia_sale.py

```diff
@@ -0,0 +1,811 @@
+    def check_control_account(self):
+        for record in self:
+            if record.account_debit_first and record.account_credit_first:
+                record.account_debit_second = record.account_credit_first.id
+                record.account_credit_second = record.account_debit_first.id
+
+
+class CampaignLineInvoice(models.Model):
+    _name = 'campaign.line.invoice'
+    _inherit = ['portal.mixin', 'mail.thread', 'mail.activity.mixin', 'utm.mixin']
+    _description = 'Line Invoice'
+    _rec_name = 'invoice_date'
+
+    campaign_id = fields.Many2one('campaign.marketing.elogia', 'Campaign', required=True, ondelete='restrict', tracking=1)
+    invoice_date = fields.Date(string='Invoice Date', required=True)
+    invoice_date_due = fields.Date(string='Due Date')
+    order_id = fields.Many2one('sale.order', 'Sale')
+    description = fields.Char('Description')
+    amount = fields.Float('Amount', tracking=1)
+    currency_id = fields.Many2one(related='campaign_id.currency_id', string="Currency")
+    company_id = fields.Many2one(related='campaign_id.company_id', string="Company")
+    invoice_id = fields.Many2one('account.move', string="Invoice", ondelete='restrict')
+    move_id = fields.Many2one('account.move', string="Move")
+    state = fields.Selection([
+        ('no_process', 'Not processed'),
+        ('process', 'Processed')
+    ], string='Status', compute='check_state_invoice', tracking=1)
+
+    def check_state_invoice(self):
+        for record in self:
+            record.state = 'no_process'
+            if record.invoice_id and record.move_id:
+                record.state = 'process'
+
+    def unlink(self):
+        for record in self:
+            if record.state == 'process':
+                raise UserError(_('You cannot delete an line invoice in the processed state.'))
+        res = super(CampaignLineInvoice, self).unlink()
+        return res
+
+    @api.onchange('invoice_id.invoice_date', 'invoice_id.invoice_date_due', 'invoice_id.ref')
+    def onchange_data_invoice(self):
+        if self.invoice_id.invoice_date:
+            self.invoice_date = self.invoice_id.invoice_date
+            if self.invoice_id.invoice_date_due:
+                self.invoice_date_due = self.invoice_id.invoice_date_due
+            if self.invoice_id.ref:
+                self.description = self.invoice_id.ref
+
+    @api.onchange('invoice_date', 'invoice_date_due')
+    def onchange_sale_order(self):
+        obj_order = self.env['sale.order'].search([('campaign_elogia_id', '=', self.campaign_id.ids[0])])
+        if self.invoice_date or self.invoice_date_due:
+            domain = {
+                'order_id': False,
+            }
+            values = {
+                'order_id': False,
+            }
+            if obj_order:
+                domain = {'order_id': [('id', 'in', obj_order.ids)]}
+            self.update(values)
+            return {'domain': domain}
```

Comment:
if not obj_order, la estructura del domain esta correcta??

Reply:
Ya este onchange no se usa.